File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/EmployeeTraining.cs
using HRManagementSystem.Domain.Enums;

namespace HRManagementSystem.Domain.Entities;
public class EmployeeTraining : BaseEntity
{
    public int EmployeeId { get; set; }
    public Employee? Employee { get; set; }

    public int TrainingCourseId { get; set; }
    public TrainingCourse? TrainingCourse { get; set; }

    public TrainingStatus Status { get; set; }
    public DateTime EnrollmentDate { get; set; } = DateTime.UtcNow;
    public DateTime? CompletionDate { get; set; }
    public bool RewardGiven { get; set; } = false;     // prevents double bonus
}

File: frontend/src/app/models/training/employee-training.ts
export type TrainingStatus = 'Enrolled' | 'Completed' | 'Cancelled';

export interface EmployeeTraining {
  employeeId: number;
  trainingCourseId: number;
  status: TrainingStatus;
  enrollmentDate: string;
  completionDate?: string;
  rewardGiven: boolean;
  employeeName?: string;
  trainingCourseTitle?: string;
}

export interface EmployeeEnrollDto {
  employeeId: number;
  trainingCourseId: number;
}
